aea/decision_maker/base.py

```diff
@@ -110,42 +111,45 @@ def check_transaction_is_consistent(self, tx_message: TransactionMessage) -> boo
         """
         Check if the transaction is consistent.
 
-        E.g. check that the agent state has enough money if it is a buyer
-        or enough holdings if it is a seller.
+        E.g. check that the agent state has enough money if it is a buyer or enough holdings if it is a seller.
+        Note, the agent is the sender of the transaction message by design.
         :return: True if the transaction is legal wrt the current state, false otherwise.
         """
-        currency_id = tx_message.currency_id
-        if tx_message.is_sender_buyer:
-            # check if we have the money to cover amount and tx fee.
-            result = self.amount_by_currency[currency_id] >= tx_message.amount + tx_message.sender_tx_fee
-        else:
-            # check if we have the goods.
-            result = True
-            for good_id, quantity in tx_message.quantities_by_good_id.items():
-                result = result and (self.quantities_by_good_id[good_id] >= quantity)
-            # check if we have the money to cover tx fee.
-            result = self.amount_by_currency[currency_id] + tx_message.amount >= tx_message.sender_tx_fee
+        result = len(tx_message.tx_amount_by_currency_id) == 1
```

Comment:
what does that mean? Do we only support one currency?

If yes, I think it's going to be a temporary state of the software. Maybe adding a comment, or a `# TODO` would help. Worth opening an issue as well.

Reply:
Well, a transaction should for the forseeable future just reference one currency. But I will add an issue to do multi currency transactions :)